### PR TITLE
[WIP] Use ogc server

### LINKED ISSUE #1752: Remove deprecated usage (facilitate qgisserver integration)
The attributes: url, isSingleTile, wfsSupport, urlWfs and serverType should be get from the ogcServer, not directly from the WMS Layer

=> should be removed (I put some others that looks strange for me ...):
- https://github.com/camptocamp/ngeo/blob/5ff1b8716c8051d4a36dc73206d7adbc312e07bb/contribs/gmf/externs/gmf-themes.js#L54
- https://github.com/camptocamp/ngeo/blob/5ff1b8716c8051d4a36dc73206d7adbc312e07bb/contribs/gmf/externs/gmf-themes.js#L66
- https://github.com/camptocamp/ngeo/blob/5ff1b8716c8051d4a36dc73206d7adbc312e07bb/contribs/gmf/externs/gmf-themes.js#L90 ? for WMTS ?
- https://github.com/camptocamp/ngeo/blob/5ff1b8716c8051d4a36dc73206d7adbc312e07bb/contribs/gmf/externs/gmf-themes.js#L101
- https://github.com/camptocamp/ngeo/blob/5ff1b8716c8051d4a36dc73206d7adbc312e07bb/contribs/gmf/externs/gmf-themes.js#L107
- https://github.com/camptocamp/ngeo/blob/5ff1b8716c8051d4a36dc73206d7adbc312e07bb/contribs/gmf/externs/gmf-themes.js#L113
- https://github.com/camptocamp/ngeo/blob/5ff1b8716c8051d4a36dc73206d7adbc312e07bb/contribs/gmf/externs/gmf-themes.js#L119

### PR DESCRIPTION
**Pull request by ger-benjamin**
First step to fix #1752 

I've added the OgcServerObject usage in the layertree (1st commit).

Then to fix the tests I've updated the contribs/gmf/test/spec/data/themes.js because it was old and don't reflect the current state of the architecture. So more tests break apart.

**I need help to fix the lasts.**

### CODE REVIEW

File: contribs/gmf/externs/gmf-themes.js [hunk 1]
```diff
@@ -150,7 +150,7 @@ GmfThemesNode.prototype.urlWfs;
 
 
 /**
- * @type {string|undefined}
+ * @type {string}
```
Comment (sbrunner) [inline contribs/gmf/externs/gmf-themes.js]: It can be undefined ...

File: contribs/gmf/src/directives/layertree.js [hunk 2]
```diff
@@ -815,9 +822,9 @@ gmf.LayertreeController.prototype.getLegendIconURL = function(treeCtrl) {
 
   //In case of multiple layers for a node, always take the first layer name to get the icon
   var layerName = node.layers.split(',')[0];
-  var url = node.url || this.gmfWmsUrl_;
-  return this.layerHelper_.getWMSLegendURL(url, layerName, this.getScale_(),
-    opt_legendRule);
+  var url = node.url + ''; // FIXME precise node type to always have an url
```
Comment (sbrunner) [inline contribs/gmf/src/directives/layertree.js]: Don't you want to fix it in this pullrequest?

**Comment by sbrunner** [comment]
Just some comment also that's looks good :-)

**Comment by ger-benjamin** [comment]
Useless with #1814

**Pull request closed without merging**